src/platform/datapath_raw_socket_win.c

```diff
@@ -315,49 +298,80 @@ CxPlatTryAddSocket(
                 "[data][%p] ERROR, %u, %s.",
                 Socket,
                 WsaError,
-                "Set SO_REUSEADDR");
+                "Set IPV6_V6ONLY");
             Status = HRESULT_FROM_WIN32(WsaError);
             goto Error;
         }
-    }
 
-    CxPlatConvertToMappedV6(&Socket->LocalAddress, &MappedAddress);
+        if (Socket->CibirIdLength) {
+            Option = TRUE;
+            Result =
+                setsockopt(
+                    Socket->AuxSocket,
+                    SOL_SOCKET,
+                    SO_REUSEADDR,
+                    (char*)&Option,
+                    sizeof(Option));
+            if (Result == SOCKET_ERROR) {
+                int WsaError = SocketError();
+                QuicTraceEvent(
+                    DatapathErrorStatus,
+                    "[data][%p] ERROR, %u, %s.",
+                    Socket,
+                    WsaError,
+                    "Set SO_REUSEADDR");
+                Status = HRESULT_FROM_WIN32(WsaError);
+                goto Error;
+            }
+        }
+
+        CxPlatConvertToMappedV6(&Socket->LocalAddress, &MappedAddress);
 #if QUIC_ADDRESS_FAMILY_INET6 != AF_INET6
-    if (MappedAddress.Ipv6.sin6_family == QUIC_ADDRESS_FAMILY_INET6) {
-        MappedAddress.Ipv6.sin6_family = AF_INET6;
-    }
+        if (MappedAddress.Ipv6.sin6_family == QUIC_ADDRESS_FAMILY_INET6) {
+            MappedAddress.Ipv6.sin6_family = AF_INET6;
+        }
 #endif
+    }
 
     CxPlatRwLockAcquireExclusive(&Pool->Lock);
 
-    Result =
-        bind(
-            Socket->AuxSocket,
-            (struct sockaddr*)&MappedAddress,
-            sizeof(MappedAddress));
-    if (Result == SOCKET_ERROR) {
-        int WsaError = SocketError();
-        QuicTraceEvent(
-            DatapathErrorStatus,
-            "[data][%p] ERROR, %u, %s.",
+    if (Socket->UseTcp)
+    {
+        QUIC_ADDR_STR LocalAddressString = {0};
+        QuicAddrToString(&MappedAddress, &LocalAddressString);
+        QuicTraceLogVerbose(
+            DatapathTcpAuxBinding,
+            "[data][%p] Binding TCP socket to %s",
             Socket,
-            WsaError,
-            "bind");
-        CxPlatRwLockReleaseExclusive(&Pool->Lock);
-        Status = HRESULT_FROM_WIN32(WsaError);
-        goto Error;
-    }
+            LocalAddressString.Address);
+        Result =
+            bind(
```

Comment:
Are we going to always try to bind to the same addresess as the normal UDP socket?

Reply:
I changed it only when QTIP enabled